app/src/main/java/com/example/habithelper/fragments/TrackFragment.java

```diff
@@ -458,7 +475,23 @@ private void initializeUserVariables(View view) {
         rbFive.setTag(5);
         currentUser = ParseUser.getCurrentUser();
         habitsList = currentUser.getList("habitsList");
-        numHabits = habitsList.size();
+        if (habitsList != null){
```

Comment:
`
 if (habitsList == null || habitsList.size() < 4 || habitsList.size() > 10) {
           goToSetup();
        }
`

Reply:
Additionally, wondering, if rather than go the setup, we should not allow continue/move to the next screen

Reply:
My idea with going to setup was that the user would be forced to add habits before being able to do anything - this way, if they were able to bypass adding habits, they could still create their profile at the start of using the app with no issues.